src/overlays/actors/ovl_Boss_01/z_boss_01.c

```diff
@@ -246,191 +648,2943 @@ ActorInit Boss_01_InitVars = {
+    }
+
+    Actor_UpdateBgCheckInfo(play, &this->actor, 50.0f, 150.0f, 100.0f, UPDBGCHECKINFO_FLAG_1 | UPDBGCHECKINFO_FLAG_4);
+    if (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
+        Math_ApproachZeroF(&this->additionalVelocityX, 1.0f, 1.0f);
+        Math_ApproachZeroF(&this->additionalVelocityZ, 1.0f, 1.0f);
+    }
+
+    this->actor.shape.rot = this->actor.world.rot;
+
+    if (this->disableCollisionTimer == 0) {
+        Boss01_UpdateDamage(this, play);
+
+        if (this->bodyInvincibilityTimer == 0) {
+            CollisionCheck_SetAC(play, &play->colChkCtx, &this->bodyCollider.base);
+        } else {
+            this->bodyInvincibilityTimer--;
+            for (i = 0; i < ODOLWA_COLLIDER_BODYPART_MAX; i++) {
+                this->bodyCollider.elements[i].info.bumperFlags &= ~BUMP_HIT;
+            }
+        }
+
+        if (this->swordAndShieldCollisionEnabled) {
+            CollisionCheck_SetAC(play, &play->colChkCtx, &this->shieldCollider.base);
+            CollisionCheck_SetOC(play, &play->colChkCtx, &this->shieldCollider.base);
+            CollisionCheck_SetAC(play, &play->colChkCtx, &this->swordCollider.base);
+        }
+
+        if (this->swordState != ODOLWA_SWORD_STATE_INACTIVE) {
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->swordCollider.base);
+        }
+
+        if (this->kickAndShieldBashCollisionEnabled) {
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->kickAndShieldBashCollider.base);
+        }
+    } else {
+        this->disableCollisionTimer--;
+        for (i = 0; i < ODOLWA_COLLIDER_BODYPART_MAX; i++) {
+            this->bodyCollider.elements[i].info.bumperFlags &= ~BUMP_HIT;
+        }
+    }
+
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->bodyCollider.base);
+
+    // If Odolwa and the player are *not* facing each other, Odolwa will *not* block or jump away from the player's
+    // attacks, even if canGuardOrEvade is set to true. This allows the player to hit Odolwa even during states where he
+    // normally evades attacks, so long as the player is far enough to the side or behind him.
+    if (this->canGuardOrEvade &&
+        ((player->unk_D57 != 0) || ((player->unk_ADC != 0) && (this->actor.xzDistToPlayer <= 120.0f))) &&
+        Boss01_ArePlayerAndOdolwaFacing(this, play)) {
+        if ((Rand_ZeroOne() < 0.25f) && (this->actionFunc != Boss01_Guard)) {
+            Boss01_SetupJump(this, play, false);
+            this->disableCollisionTimer = 10;
+        } else if ((player->unk_ADC != 0) && (this->actor.xzDistToPlayer <= 120.0f)) {
+            Boss01_SetupGuard(this, play, true);
+        } else {
+            Boss01_SetupGuard(this, play, false);
+        }
+    }
+
+    if (this->canGuardOrEvade) {
+        Boss01_JumpAwayFromExplosive(this, play);
+    }
+
+    if (((this->frameCounter & (this->afterimageSpawnFrameMask - 1)) == 0) && (this->afterimageSpawnFrameMask != 0)) {
```

Comment:
```suggestion
    if ((((this->frameCounter != 0) & (this->afterimageSpawnFrameMask - 1)) == 0) && (this->afterimageSpawnFrameMask != 0)) {
```

Reply:
Are you sure about this one? This is a bitwise `&`, not an `&&`.

Reply:
Oh yeah you're right, sorry. Read that as a `&` instead of an `&&`

Reply:
No worries, that `(this->afterimageSpawnFrameMask - 1)` makes this frankly super awkward to read, so it's an easy mistake to make.